pkg/controller/s3/bucket/lifecycleConfig.go

```diff
@@ -163,3 +165,13 @@ func (in *LifecycleConfigurationClient) Delete(ctx context.Context, bucket *v1be
 	).Send(ctx)
 	return errors.Wrap(err, lifecycleDeleteFailed)
 }
+
+func sortFilterTags(rules []awss3.LifecycleRule) {
+	if len(rules) != 0 {
```

Comment:
If `len(rules) == 0`, `for` loop wouldn't run at all so the function would be no-op. So, I think we can remove that `if` check here.

Reply:
This was mainly here since `len(nil)` would also resolve to 0, but I have changed this to something more clear.